common/components/BalanceSidebar/Promos.tsx

```diff
@@ -94,6 +58,15 @@ export default class Promos extends React.Component<{}, State> {
   }
 
   private navigateToPromo = (idx: number) => () => {
+    // stop rotating when user begins interacting with promos
+    clearInterval(this.timer);
     this.setState({ activePromo: Math.max(0, Math.min(promos.length, idx)) });
   };
+
+  private rotate = () => {
+    const previousIndex = this.state.activePromo;
+    const nextIndex = previousIndex + 1;
+    const nextIndexExists = typeof promos[nextIndex] !== 'undefined' ? true : false;
```

Comment:
You can make this code _much_ simpler with modular math:
```
const activePromo = (this.state.activePromo + 1) % promos.length;
this.setState({ activePromo });
```
The % sign is a modulo operator, which causes numbers to wrap.

Reply:
Good use of modulo 👍 , thx for the suggestion